### PR TITLE
tidb-lightning: max-error and duplicate-resolution

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Document the error resolution features, introduced in pingcap/tidb#27181 and pingcap/tidb#28464.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/6717
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/tidb-lightning-configuration.md [hunk 1]
```diff
@@ -59,6 +59,15 @@ table-concurrency = 6
 # 对于不同的存储介质，此参数可能需要调整以达到最佳效率。
 io-concurrency = 5
 
+# TiDB Lightning 能够承受的最大非致命（non-fatal）错误数。超过此数，TiDB Lightning 会停止任务。
+# 非致命（non-fatal）错误指的是只出现在某些行中的错误。忽略这些错误可以让导入进程继续下去。
+# 如果将该值设置为 N，那么 TiDB Lightning 会在遇到第 N+1 个非致命错误时即可停止。
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:64]:
```suggestion
# 如果将该值设置为 N，那么 TiDB Lightning 会在遇到第 N+1 个非致命错误时停止。
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 2]
```diff
@@ -102,6 +111,12 @@ addr = "172.16.31.10:8287"
 # - ignore：保留已有数据，忽略新数据
 # - error：中止导入并报错
 # on-duplicate = "replace"
+# 当后端模式为 'local' 时，设置是否检测和解决重复的记录（唯一键冲突）。
+# 目前支持三种解决方法：
+#  - record: 仅将重复记录添加到目的 TiDB 中的 `lightning_task_info.conflict_error_v1` 表中。注意，该方法要求目的 TiKV 的版本为 v5.2.0 或更新版本。如果版本过低，则会启用下面的 'none' 模式。
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-configuration.md:116]:
```suggestion
#  - record: 仅将重复记录添加到目标 TiDB 中的 `lightning_task_info.conflict_error_v1` 表中。注意，该方法要求目标 TiKV 的版本为 v5.2.0 或更新版本。如果版本过低，会退化成下面的 'none' 模式。
```

not sure if "退化" is a proper translation of "fallback", but "启用" definitely is not.

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+```toml
+[lightning]
+max-error = 0
+```
+
+该配置对下列错误有效：
+
+* 无效值。例如：在 INT 列设置了 `'Text'`
+* 数字溢出。例如：在 TINYINT 列设置了 500
+* 字符串溢出。例如: 在 VARCHAR(5) column 列中设置了`'Very Long Text'`
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-error-resolution.md:33]:
```suggestion
* 字符串溢出。例如: 在 VARCHAR(5) 列中设置了`'非常长的文字'`
```

File: tidb-lightning/tidb-lightning-configuration.md (hunk 2 above)
Comment (hfxsd) [reply tidb-lightning/tidb-lightning-configuration.md:116]: 好像用 “回退” 比较好一些？

Comment (kennytm) [reply tidb-lightning/tidb-lightning-configuration.md:116]: 回退 +1

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+title: TiDB Lightning 错误处理
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-error-resolution.md:2]: 这个标题，可以改成 “TiDB Lightning 容错处理” 吗？

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+summary: 介绍了如何解决导入数据过程中的类型转换和冲突错误。
+---
+
+# 错误处理
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-error-resolution.md:6]: 建议和 TOC 中的名字保持一致。另外只看标题，可能以为是遇到导入错误了，要如何处理，但文中的内容更偏向“错误检测机制”。

File: TOC.md [hunk 6]
```diff
@@ -167,6 +167,7 @@
       + [表库过滤](/table-filter.md)
       + [CSV 支持](/tidb-lightning/migrate-from-csv-using-tidb-lightning.md)
       + [TiDB Lightning后端](/tidb-lightning/tidb-lightning-backends.md)
+      + [故障处理](/tidb-lightning/tidb-lightning-error-resolution.md)
```
Comment (hfxsd) [inline TOC.md:170]:
```suggestion
      + [错误处理](/tidb-lightning/tidb-lightning-error-resolution.md)
```

File: tidb-lightning/tidb-lightning-error-resolution.md (hunk 5 above)
Comment (hfxsd) [reply tidb-lightning/tidb-lightning-error-resolution.md:6]: 现在这个标题名字确实有点宽泛。我们考虑下怎么改。

File: tidb-lightning/tidb-lightning-error-resolution.md (hunk 4 above)
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-error-resolution.md:2]:
```suggestion
title: TiDB Lightning 错误处理配置
```

File: tidb-lightning/tidb-lightning-error-resolution.md (hunk 5 above)
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-error-resolution.md:6]:
```suggestion
# 错误处理配置
```

File: TOC.md [hunk 7]
```diff
@@ -167,6 +167,7 @@
       + [表库过滤](/table-filter.md)
       + [CSV 支持](/tidb-lightning/migrate-from-csv-using-tidb-lightning.md)
       + [TiDB Lightning后端](/tidb-lightning/tidb-lightning-backends.md)
+      + [错误处理](/tidb-lightning/tidb-lightning-error-resolution.md)
```
Comment (hfxsd) [inline TOC.md:170]:
```suggestion
      + [错误处理配置](/tidb-lightning/tidb-lightning-error-resolution.md)
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+| raw_key      |        |      | ✓        | 冲突的 KV 对的键                                                           |
+| raw_value    |        |      | ✓        | 冲突的 KV 对的值                                                            |
+| raw_handle   |        |      | ✓        | 冲突行的行句柄                                                         |
+| raw_row      |        |      | ✓        | 冲突行的编码值                                                       |
+
+> **注意：**
+>
+> 错误报告记录的是文件偏移量，不是行号或列号，因为行号或列号的获取效率很低。你可以使用下列命令在字节位实现快速跳转（以 183 为例）：
+>
+> * shell, printing several lines before —
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:141]:
```suggestion
> * shell, printing several lines before —
```
这个还不知道怎么翻

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+    raw_value   mediumblob NOT NULL,
+    raw_handle  mediumblob NOT NULL,
+    raw_row     mediumblob NOT NULL,
+    KEY (task_id, table_name)
+);
+```
+
+**syntax_error_v1** 记录文件中的语法错误。目前尚未生效。
+
+**type_error_v1** 记录由 `max-error` 配置项管理的所有[类型错误 (Type error)](#类型错误-Type-error)。每个错误一行。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:116]:
```suggestion
**type_error_v1** 记录由 `max-error` 配置项管理的所有[类型错误 (Type error)](#类型错误-type-error)。每个错误一行。
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+
+该配置对下列错误有效：
+
+* 无效值。例如：在 INT 列设置了 `'Text'`
+* 数字溢出。例如：在 TINYINT 列设置了 500
+* 字符串溢出。例如: 在 VARCHAR(5) 列中设置了`'非常长的文字'`
+* 零日期时间，如 `'0000-00-00'` 和 `'2021-12-00'`
+* 在 NOT NULL 列中设置了 NULL
+* 生成的列表达式求值失败
+* 列计数不匹配。行中数值的数量和列的数量不一致。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:37]:
```suggestion
* 列计数不匹配。行中数值的数量和列的数量不一致
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+* `on-duplicate = "error"` 时，TiDB 后端的唯一键/主键冲突
+* 其他 SQL 错误
+
+下列错误是致命错误，不能通过配置 `max-error` 跳过：
+
+* 原始 CSV、SQL 或者 Parquet 文件中的语法错误，例如未闭合的引号。
+* I/O、网络、或系统权限错误。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:44]:
```suggestion
* 原始 CSV、SQL 或者 Parquet 文件中的语法错误，例如未闭合的引号
* I/O、网络、或系统权限错误
```

**Review by Liuxiaozhen12 — COMMENTED** [review]
rest LGTM

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+    raw_row     mediumblob NOT NULL,
+    KEY (task_id, table_name)
+);
+```
+
+**syntax_error_v1** 记录文件中的语法错误。目前尚未生效。
+
+**type_error_v1** 记录由 `max-error` 配置项管理的所有[类型错误 (Type error)](#类型错误-type-error)。每个错误一行。
+
+**conflict_error_v1** 记录所有[后端中的唯一键/主键冲突](#解决重复问题)。每对冲突有 2 行。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:118]:
```suggestion
**conflict_error_v1** 记录所有[后端中的唯一键/主键冲突](#解决重复问题)。每对冲突有两行。
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+* 列计数不匹配。行中数值的数量和列的数量不一致。
+* `on-duplicate = "error"` 时，TiDB 后端的唯一键/主键冲突
+* 其他 SQL 错误
+
+下列错误是致命错误，不能通过配置 `max-error` 跳过：
+
+* 原始 CSV、SQL 或者 Parquet 文件中的语法错误，例如未闭合的引号。
+* I/O、网络、或系统权限错误。
+
+在 Local 后端模式下，唯一键/主键冲突的冲突是单独处理的。将在接下来的章节进行介绍。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-error-resolution.md:46]:
```suggestion
在 Local 后端模式下，唯一键/主键的冲突是单独处理的。将在接下来的章节进行介绍。
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+        (40, 'forty'),          -- conflicts with the other 40 below
+        (54, 'fifty-four'),     -- conflicts with the other 'fifty-four' below
+        (77, 'seventy-seven'),  -- the string is longer than 12 characters
+        (600, 'six hundred'),   -- the number overflows TINYINT
+        (40, 'fourty'),         -- conflicts with the other 40 above
+        (42, 'fifty-four');     -- conflicts with the other 'fifty-four' above
+    EOF
+    ```
+
+3. 配置 TiDB Lightning，启用严格 SQL 模式，使用本地后端进行导入，通过删除解决重复项，并最多跳过 10 个错误：
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-error-resolution.md:188]:
```suggestion
3. 配置 TiDB Lightning，启用严格 SQL 模式，使用 Local 后端模式进行导入，通过删除解决重复项，并最多跳过 10 个错误：
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+>
+> 错误报告记录的是文件偏移量，不是行号或列号，因为行号或列号的获取效率很低。你可以使用下列命令在字节位实现快速跳转（以 183 为例）：
+>
+> * shell, printing several lines before —
+>
+>     ```sh
+>     head -c 183 file.csv | tail
+>     ```
+>
+> * shell, printing several lines after —
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:147]:
```suggestion
> * shell, 输出一些后置的数据行——
```

**Review by Liuxiaozhen12 — APPROVED** [review]

File: tidb-lightning/tidb-lightning-configuration.md [hunk 16]
```diff
@@ -102,6 +111,12 @@ addr = "172.16.31.10:8287"
...
+#  - none: 不检测重复记录。该模式是三种模式中性能最佳的，但是可能会导致目的 TiDB 中出现数据不一致的情况。
+#  - remove: 记录所有的重复记录，和'record'模式相似。但是会删除所有的重复记录，以确保目的 TiDB 中的数据状态保持一致。
+# duplicate-resolution = 'record'
```
Comment (glorv) [inline tidb-lightning/tidb-lightning-configuration.md:119]:
```suggestion
# duplicate-resolution = 'none'
```

Comment (glorv) [reply tidb-lightning/tidb-lightning-configuration.md:119]: 默认值调整为 none 了

File: TOC.md [hunk 17]
```diff
@@ -167,6 +167,7 @@
       + [表库过滤](/table-filter.md)
       + [CSV 支持](/tidb-lightning/migrate-from-csv-using-tidb-lightning.md)
       + [TiDB Lightning后端](/tidb-lightning/tidb-lightning-backends.md)
+      + [错误处理配置](/tidb-lightning/tidb-lightning-error-resolution.md)
```
Comment (glorv) [inline TOC.md:170]:
```suggestion
      + [错误处理](/tidb-lightning/tidb-lightning-error-resolution.md)
```

**Review by glorv — COMMENTED** [review]
rest LGTM

**Review by sunzhaoyang — APPROVED** [review]
LGTM

File: TOC.md (hunk 17 above)
Comment (Liuxiaozhen12) [inline TOC.md]:
```suggestion
      + [错误处理](/tidb-lightning/tidb-lightning-error-resolution.md)
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
+---
+title: TiDB Lightning 错误处理配置
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:2]:
```suggestion
title: TiDB Lightning 错误处理功能
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+summary: 介绍了如何解决导入数据过程中的类型转换和冲突错误。
+---
+
+# 错误处理配置
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:6]:
```suggestion
# TiDB Lightning 错误处理功能
```

File: tidb-lightning/tidb-lightning-configuration.md [hunk 20]
```diff
@@ -102,6 +111,12 @@ addr = "172.16.31.10:8287"
 # - ignore：保留已有数据，忽略新数据
 # - error：中止导入并报错
 # on-duplicate = "replace"
+# 当后端模式为 'local' 时，设置是否检测和解决重复的记录（唯一键冲突）。
+# 目前支持三种解决方法：
+#  - record: 仅将重复记录添加到目的 TiDB 中的 `lightning_task_info.conflict_error_v1` 表中。注意，该方法要求目的 TiKV 的版本为 v5.2.0 或更新版本。如果版本过低，则会启用下面的 'none' 模式。
+#  - none: 不检测重复记录。该模式是三种模式中性能最佳的，但是可能会导致目的 TiDB 中出现数据不一致的情况。
+#  - remove: 记录所有的重复记录，和'record'模式相似。但是会删除所有的重复记录，以确保目的 TiDB 中的数据状态保持一致。
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-configuration.md:118]:
```suggestion
#  - remove: 记录所有的重复记录，和 'record' 模式相似。但是会删除所有的重复记录，以确保目的 TiDB 中的数据状态保持一致。
```

File: TOC.md (hunk 7 above)
Comment (Liuxiaozhen12) [inline TOC.md:170]:
```suggestion
      + [错误处理功能](/tidb-lightning/tidb-lightning-error-resolution.md)
```

File: TOC.md [hunk 21]
```diff
@@ -167,6 +167,7 @@
       + [表库过滤](/table-filter.md)
       + [CSV 支持](/tidb-lightning/migrate-from-csv-using-tidb-lightning.md)
       + [TiDB Lightning后端](/tidb-lightning/tidb-lightning-backends.md)
```
Comment (Liuxiaozhen12) [inline TOC.md:169]:
```suggestion
      + [TiDB Lightning 后端](/tidb-lightning/tidb-lightning-backends.md)
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 22]
```diff
@@ -0,0 +1,309 @@
...
+
+## 解决重复问题
+
+Local 后端模式下，TiDB Lightning 导入数据时先将数据转换成 KV 对数组（KV pairs），然后批量添加到 TiKV 中。与 TiDB 后端模式不同，直到任务结束才会检测重复行。因此，Local 模式下的重复错误不是通过 `max-error` 进行控制，而是通过 `duplicate-resolution` 配置项进行控制的。
+
+{{< copyable "" >}}
+
+```toml
+[tikv-importer]
+duplicate-resolution = 'record'
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:56]: @glorv 请问这里需要调整为默认值吗？
```suggestion
duplicate-resolution = 'record'
```

Comment (kennytm) [reply tidb-lightning/tidb-lightning-error-resolution.md:56]: Should be changed to `none`.

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 23]
```diff
@@ -0,0 +1,309 @@
...
+| raw_key      |        |      | ✓        | 冲突的 KV 对的键                                                           |
+| raw_value    |        |      | ✓        | 冲突的 KV 对的值                                                            |
+| raw_handle   |        |      | ✓        | 冲突行的行句柄                                                         |
+| raw_row      |        |      | ✓        | 冲突行的编码值                                                       |
+
+> **注意：**
+>
+> 错误报告记录的是文件偏移量，不是行号或列号，因为行号或列号的获取效率很低。你可以使用下列命令在字节位实现快速跳转（以 183 为例）：
+>
+> * shell, 输出一些前置的数据行——
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-error-resolution.md:141]:
```suggestion
> * shell, 输出前面几行——
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 24]
```diff
@@ -0,0 +1,309 @@
...
+>
+>     ```sh
+>     head -c 183 file.csv | tail
+>     ```
+>
+> * shell, 输出一些后置的数据行——
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-error-resolution.md:147]:
```suggestion
> * shell, 输出后面几行——
```

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 25]
```diff
@@ -0,0 +1,309 @@
...
+        INSERT INTO t (a, b) VALUES
+        (0, NULL),              -- column is NOT NULL
+        (1, 'one'),
+        (2, 'two'),
+        (40, 'forty'),          -- conflicts with the other 40 below
+        (54, 'fifty-four'),     -- conflicts with the other 'fifty-four' below
+        (77, 'seventy-seven'),  -- the string is longer than 12 characters
+        (600, 'six hundred'),   -- the number overflows TINYINT
+        (40, 'fourty'),         -- conflicts with the other 40 above
+        (42, 'fifty-four');     -- conflicts with the other 'fifty-four' above
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-error-resolution.md:184]: translate the comments...

**Review by kennytm — COMMENTED** [review]
rest LGTM

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 26]
```diff
@@ -0,0 +1,309 @@
...
+        INSERT INTO t (a, b) VALUES
+        (0, NULL),              -- 列不为空
+        (1, 'one'),
+        (2, 'two'),
+        (40, 'forty'),          -- 与下面的 `40` 冲突
+        (54, 'fifty-four'),     -- 与下面的 `54` 冲突
+        (77, 'seventy-seven'),  -- 字符串长度超过 12 个字符
+        (600, 'six hundred'),   -- 数字超出了 TINYINT 数据类型支持的范围
+        (40, 'fourty'),         -- 与上面的 `40` 冲突
+        (42, 'fifty-four');     -- 与上面的 `54` 冲突
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:184]:
```suggestion
        (42, 'fifty-four');     -- 与上面的 `54` 冲突
```
done~ PTAL. And I wonder if the number "42" should be "54"? @kennytm

Comment (kennytm) [reply tidb-lightning/tidb-lightning-error-resolution.md:184]: @Liuxiaozhen12  42 is correct.

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 27]
```diff
@@ -0,0 +1,309 @@
...
+    {{< copyable "shell-regular" >}}
+
+    ```sh
+    cat <<EOF > example.t.1.sql
+        INSERT INTO t (a, b) VALUES
+        (0, NULL),              -- 列不为空
+        (1, 'one'),
+        (2, 'two'),
+        (40, 'forty'),          -- 与下面的 `40` 冲突
+        (54, 'fifty-four'),     -- 与下面的 `54` 冲突
```
Comment (kennytm) [inline tidb-lightning/tidb-lightning-error-resolution.md:180]:
```suggestion
        (54, 'fifty-four'),     -- 与下面的 `'fifty-four'` 冲突
```

File: tidb-lightning/tidb-lightning-error-resolution.md (hunk 26 above)
Comment (kennytm) [reply tidb-lightning/tidb-lightning-error-resolution.md:184]:
```suggestion
        (42, 'fifty-four');     -- 与上面的 `'fifty-four'` 冲突
```

**Review by kennytm — APPROVED** [review]

File: tidb-lightning/tidb-lightning-error-resolution.md [hunk 28]
```diff
@@ -0,0 +1,309 @@
...
+Local 后端模式下，TiDB Lightning 导入数据时先将数据转换成 KV 对数组（KV pairs），然后批量添加到 TiKV 中。与 TiDB 后端模式不同，直到任务结束才会检测重复行。因此，Local 模式下的重复错误不是通过 `max-error` 进行控制，而是通过 `duplicate-resolution` 配置项进行控制的。
+
+{{< copyable "" >}}
+
+```toml
+[tikv-importer]
+duplicate-resolution = 'none'
+```
+
+`duplicate-resolution`有以下三个选项：
```
Comment (Liuxiaozhen12) [inline tidb-lightning/tidb-lightning-error-resolution.md:59]:
```suggestion
`duplicate-resolution` 有以下三个选项：
```

**Pull request merged**